### ISSUE TITLE
Requirements for masks

### ISSUE THREAD

**Issue by smith6jt-cop** [murphygroup/CellSegmentationEvaluator#3]
Your instructions indicate a requirement of two channels for the masks:

The masks should be in a similar format with an indexed image for cell masks in the first channel and an indexed image (with corresponding indices) for nuclear masks in the second channel.

However, your example 2D_CODEX.ome.tiff masks is an image with 4 channels.  These masks do not appear to be typical binary/labeled masks produced by any typical imaging platform.

Please provide clear complete instructions for generating the appropriate inputs.

**Comment by rfmurphy** [comment]
The mask image should contain channels for a cell mask and a nucleus mask (in that order).  Other mask channels may be present but will be ignored.  Each channel should contain an indexed image, in which pixels contain the integer number of the cell that that pixel belongs to.  Most cell segmentation tools produce an indexed image.

I will be updating the README shortly.

**Comment by smith6jt-cop** [comment]
Does the mask need to be formatted such that there is a grayscale gradient that begins at the top of the image and then gradually decreases to the bottom like in the example mask image?  This is the aspect that seems atypical.  Does the gradient itself create the integer values for the mask so the bit depth is the bounds for the maximum number of cells?

**Comment by rfmurphy** [comment]
Dear Justin:

I’m debugging with your images that you sent me links to on December 4.  Using that mask file, aicsimageio is returning the mask as uint16, and more than one cell has the same value (see below).  Those files also read with the masks in the z slices like I mentioned earlier.

You mentioned in a later email that you “fixed” the images using skimage.io and posted updated images.  However, I don’t see them anywhere.  Can you send me links to the “fixed” images?  I suspect that is the source of our disconnect.

I’ll make it a priority to debug this once I get the new images.

Best,

Bob

mp.imshow(mask1)
￼
mp.imshow(mask1==5)
￼
=======================
Robert F. Murphy, Ph.D.
Ray and Stephanie Lane Professor of Computational Biology Emeritus
School of Computer Science
Carnegie Mellon University
email: ***@***.***
AIMBE Fellow, IEEE Fellow

**Comment by smith6jt-cop** [comment]
Hello,

Thank you for your help.  I cannot see the output for imshow; only OBJ with a dotted line around it.  Here is the link to the 32bit mask image:

https://drive.google.com/file/d/1AUa32zqsywUK-sDBci6XORNUH2xnfj9r/view?usp=sharing


Thanks,

Justin Smith, PhD
Dept of Pathology, Immunology, and Laboratory Medicine
UF Diabetes Institute
College of Medicine
1275 Center Drive, 5th Floor
Gainesville, FL 32610

**Comment by smith6jt-cop** [comment]
I found that there are known issues with labeled mask images created in QuPath using .useInstanceLabels().  See https://forum.image.sc/t/exporting-cell-segmentation-as-label-mask/66999/9 .  However, I believe the problem has been fixed in the most recent version of QuPath-0.5 as .useInstanceLabels() results in mask images similar to the example mask images (grayscale gradient).  

The issue regarding whether the requirement for mask images is 4 channels as the example mask images have has been resolved.
